tools/cli/go-whisk/whisk/api.go

```diff
@@ -0,0 +1,272 @@
+ * limitations under the License.
+ */
+
+package whisk
+
+import (
+    "fmt"
+    "net/http"
+    "errors"
+    "net/url"
+    "../wski18n"
+)
+
+type ApiService struct {
+    client *Client
+}
+
+type SendApi struct {
+    ApiDoc         *Api       `json:"apidoc,omitempty"`
+}
+
+type Api struct {
+    Namespace       string    `json:"namespace,omitempty"`
+    ApiName         string    `json:"apiName,omitempty"`
+    GatewayBasePath string    `json:"gatewayBasePath,omitempty"`
+    GatewayRelPath  string    `json:"gatewayPath,omitempty"`
+    GatewayMethod   string    `json:"gatewayMethod,omitempty"`
+    Id              string    `json:"id,omitempty"`
+    GatewayFullPath string    `json:"gatewayFullPath,omitempty"`
+    Swagger         string    `json:"swagger,omitempty"`
+    Action          *ApiAction `json:"action,omitempty"`
+}
+
+type ApiAction struct {
+    Name            string    `json:"name,omitempty"`
+    Namespace       string    `json:"namespace,omitempty"`
+    BackendMethod   string    `json:"backendMethod,omitempty"`
+    BackendUrl      string    `json:"backendUrl,omitempty"`
+    Auth            string    `json:"authkey,omitempty"`
+}
+
+type ApiOptions struct {
+    ActionName      string    `url:"action,omitempty"`
+    ApiBasePath     string    `url:"basepath,omitempty"`
+    ApiRelPath      string    `url:"relpath,omitempty"`
+    ApiVerb         string    `url:"operation,omitempty"`
+    ApiName         string    `url:"apiname,omitempty"`
+    Force           bool      `url:"force,omitempty"`
+}
+
+type ApiListOptions struct {
+    ApiOptions
+    Limit           int       `url:"limit"`
+    Skip            int       `url:"skip"`
+    Docs            bool      `url:"docs,omitempty"`
+}
+
+type RetApiReponse struct {
+    Response        *RetResult `json:"response"`
+}
+
+type RetResult struct {
+    Result          *RetApi   `json:"result"`
+}
+
+type RetApiReponseApiArray struct {
+    Response        *RetResultApiArray `json:"response"`
+}
+
+type RetResultApiArray struct {
+    ResultArray     *RetApiArray `json:"result"`
+}
+
+type RetApiArray struct {
+    Apis            []ApiItem `json:"apis,omitempty"`
+}
+
+type ApiItem struct {
+    ApiId           string    `json:"id,omitempty"`
+    QueryKey        string    `json:"key,omitempty"`
+    ApiValue        *RetApi   `json:"value,omitempty"`
+}
+
+type RetApi struct {
+    Namespace       string    `json:"namespace"`
+    BaseUrl         string    `json:"gwApiUrl"`
+    Activated       bool      `json:"gwApiActivated"`
+    TenantId        string    `json:"tenantId"`
+    Swagger         *ApiSwagger `json:"apidoc,omitempty"`
+}
+
+type ApiSwagger struct {
+    SwaggerName     string    `json:"swagger,omitempty"`
+    BasePath        string    `json:"basePath,omitempty"`
+    Info            *ApiSwaggerInfo `json:"info,omitempty"`
+    Paths           map[string]map[string]map[string]map[string]interface{} `json:"paths,omitempty"`
```

Comment:
@dubeejw - hopefully there's a more readable way to dig into a nest JSON without having to define structures for every intermediate object in the nested path.